src/util/factorymanager.cpp

```diff
@@ -89,33 +162,90 @@ FactoryManager& FactoryManager::ref() {
 }
 
 std::string FactoryManager::generateJson() const {
-    std::stringstream json;
+    nlohmann::json json;
 
-    json << "[";
     for (const FactoryInfo& factoryInfo : _factories) {
-        json << "{";
-        json << "\"name\": \"" << factoryInfo.name << "\",";
-        json << "\"classes\": [";
+        nlohmann::json factory;
+        factory["Name"] = factoryInfo.name;
 
         ghoul::TemplateFactoryBase* f = factoryInfo.factory.get();
         const std::vector<std::string>& registeredClasses = f->registeredClasses();
         for (const std::string& c : registeredClasses) {
-            json << "\"" << c << "\"";
-            if (&c != &registeredClasses.back()) {
-                json << ",";
-            }
+            json["Classes"].push_back(c);
+        }
+        json["Data"].push_back(factory);
+    }
+
+    // I did not check the output of this for correctness ---abock
```

Comment:
Is this comment still correct?  I'm assuming you checked it? :D

Reply:
I checked the new documentation with the old one, yes. But I didn't check if the old one was correct